### PR TITLE
Include dSYMS when archiving built frameworks

### PR DESCRIPTION
**Pull request by sharplet**
:v:

### CODE REVIEW

**Comment by sharplet** [comment]
Example output:

```
*** Found Carthage/Build/Mac/Foo.framework
*** Found Carthage/Build/Mac/Foo.framework.dSYM
*** Found Carthage/Build/iOS/Foo.framework
*** Found Carthage/Build/iOS/Foo.framework.dSYM
*** Created Foo.framework.zip
```

File: Source/carthage/Archive.swift [hunk 1]
```diff
@@ -23,6 +23,9 @@ public struct ArchiveCommand: CommandType {
 
 				return SignalProducer(values: Platform.supportedPlatforms)
 					|> map { platform in platform.relativePath.stringByAppendingPathComponent(options.frameworkName).stringByAppendingPathExtension("framework")! }
+					|> flatMap(.Concat) { relativePath in
+						return SignalProducer(values: [relativePath, relativePath.stringByAppendingPathExtension("dSYM")!])
```
Comment (mdiep) [inline Source/carthage/Archive.swift]: Could you combine this with the previous `map` so it's done in one step instead of 2?

**Comment by mdiep** [comment]
Works great—thanks for opening a pull request! :sparkles: I just added one note about the code.

**Comment by sharplet** [comment]
Yep, done.

**Comment by mdiep** [comment]
Awesome. Thanks! :sparkles:

**Pull request merged**